ecl2df/vfp.py

```diff
@@ -0,0 +1,1446 @@
+            "Dimensions of interpolation ranges "
+            "does not match number of tabulated records"
+        )
+
+    # Extract interpolation values and tabulated values (BHP values)
+    bhp_table_values: List[List[float]] = []
+    thp_values_list: List[float] = []
+    wfr_values_list: List[float] = []
+    gfr_values_list: List[float] = []
+    alq_values_list: List[float] = []
+    for n in range(6, num_rec):
+        bhp_record = common.parse_opmio_deckrecord(keyword[n], "VFPPROD", "records", 6)
+        bhp_values: Union[Any, List[float]]
+        if isinstance(bhp_record.get("VALUES"), list):
+            bhp_values = bhp_record.get("VALUES")
+        elif isinstance(bhp_record.get("VALUES"), numbers.Number):
+            bhp_values = [bhp_record.get("VALUES")]
+
+        thp_index = bhp_record["THP_INDEX"] - 1
+        wfr_index = bhp_record["WFR_INDEX"] - 1
+        gfr_index = bhp_record["GFR_INDEX"] - 1
+        alq_index = bhp_record["ALQ_INDEX"] - 1
+
+        thp_value = thp_values[thp_index]
+        wfr_value = wfr_values[wfr_index]
+        gfr_value = gfr_values[gfr_index]
+        alq_value = alq_values[alq_index]
+
+        thp_values_list.append(thp_value)
+        wfr_values_list.append(wfr_value)
+        gfr_values_list.append(gfr_value)
+        alq_values_list.append(alq_value)
+
+        if len(bhp_values) != no_flow_values:
+            raise ValueError(
+                "Dimension of record of tabulated "
+                "values does not match number of flow values"
+            )
+        bhp_table_values.append(bhp_values)
+
+    # create stacked dataframe from VFP table values
+    index_names = ["PRESSURE", "WFR", "GFR", "ALQ"]
+    index_values = [thp_values_list, wfr_values_list, gfr_values_list, alq_values_list]
+    df_bhp_stacked = vfptable2df(
+        index_names, index_values, flow_values, bhp_table_values
+    )
+
+    # Add meta-data
+    df_bhp_stacked["VFP_TYPE"] = "VFPPROD"
+    df_bhp_stacked["TABLE_NUMBER"] = table
+    df_bhp_stacked["DATUM"] = datum
+    df_bhp_stacked["UNIT_TYPE"] = units.value
+    df_bhp_stacked["RATE_TYPE"] = rate.value
+    df_bhp_stacked["WFR_TYPE"] = wfr.value
+    df_bhp_stacked["GFR_TYPE"] = gfr.value
+    df_bhp_stacked["ALQ_TYPE"] = alq.value
+    df_bhp_stacked["PRESSURE_TYPE"] = thp.value
+    df_bhp_stacked["TAB_TYPE"] = tab.value
+
+    # Sort the columns in wanted order
+    df_bhp_stacked = df_bhp_stacked[
+        [
+            "RATE",
+            "PRESSURE",
+            "WFR",
+            "GFR",
+            "ALQ",
+            "TAB",
+            "VFP_TYPE",
+            "TABLE_NUMBER",
+            "DATUM",
+            "RATE_TYPE",
+            "WFR_TYPE",
+            "GFR_TYPE",
+            "ALQ_TYPE",
+            "PRESSURE_TYPE",
+            "TAB_TYPE",
+            "UNIT_TYPE",
+        ]
+    ]
+
+    # reset index (not used other than tests)
+    df_bhp_stacked.reset_index(inplace=True, drop=True)
+    return df_bhp_stacked
```

Comment:
This is details, but maybe more elegant to write:

`return df_bhp_stacked.reset_index(drop=True)`